cli/src/pcluster/models/cluster_resources.py

```diff
@@ -74,6 +74,22 @@ def batch_compute_environment(self):
         """Return Batch compute environment."""
         return self._get_output("BatchComputeEnvironmentArn")
 
+    def get_failure_reason(self):
+        """Reason of the failure when the cluster_status is in CREATE_FAILED."""
+        stack_events = get_all_stack_events(self.name)
+        wait_conditiaon_failure_events = None
+        for chuck in stack_events:
+            if wait_conditiaon_failure_events:
+                break
+            for event in chuck:
+                if (
+                    event.get("ResourceType") == "AWS::CloudFormation::WaitCondition"
+                    and event.get("ResourceStatus") == "CREATE_FAILED"
+                ):
+                    wait_conditiaon_failure_events = event
```

Comment:
Typo in `wait_conditiaon_failure_events`.

Reply:
What about just returning when you doing the right event?

Reply:
Fixed the typo.
>What about just returning when you doing the right event?
Good idea!